### PR TITLE
Sol keyboard conversion to split common

### PR DESCRIPTION
**Pull request by XScorpion2**
## Description

Update the Sol from RGBKB to use Split Common

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by kageurufu** [comment]
I've verified this works on my Sol.

**Comment by XScorpion2** [comment]
Besides my complete failure to test check in code that actually compiles tonight (works locally...go figure), apparently split common has issues with the slave side oled driver. Will fix it in the morning.

**Comment by XScorpion2** [comment]
Alright, fixed the compile errors and oled slave issues. Good to go!

**Comment by XScorpion2** [comment]
Based on feedback from Pelrun, hoisted oled code from quantum.c to keyboard.c

**Comment by yanfali** [comment]
@Legonut can you please review these changes? Thanks

**Comment by Legonut** [comment]
@yanfali yep, all good!

File: keyboards/sol/config.h [hunk 1]
```diff
@@ -20,3 +20,75 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 #include "config_common.h"
 
+#define PREVENT_STUCK_MODIFIERS
```
Comment (drashna) [inline keyboards/sol/config.h]: PSM is depreciated, as it's the default behavior. Please remove. 

```suggestion
```

File: keyboards/sol/config.h [hunk 2]
```diff
@@ -20,3 +20,75 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
+#define TAPPING_FORCE_HOLD
```
Comment (drashna) [inline keyboards/sol/config.h]: Cries in One Shot

Comment (XScorpion2) [reply keyboards/sol/config.h]: this was the default sol behavior, just restructured the directory a bit in prop for r2

File: keyboards/sol/config.h (hunk 1 above)
Comment (XScorpion2) [reply keyboards/sol/config.h]: Fixed

File: keyboards/sol/config.h (hunk 2 above)
Comment (drashna) [reply keyboards/sol/config.h]: That's fine.  It just breaks one shots, tap toggle, and some other functionality.

Comment (XScorpion2) [reply keyboards/sol/config.h]: I'll ask @Legonut about removing it.

**Comment by drashna** [comment]
Thanks

**Pull request merged**